src/main/java/org/jabref/gui/EntryTypeView.java

```diff
@@ -0,0 +1,291 @@
+    @FXML private TitledPane customPane;
+    @FXML private VBox vBox;
+
+    private final ControlsFxVisualizer visualizer = new ControlsFxVisualizer();
+    private final BasePanel basePanel;
+    private EntryType type;
+    private Task<Optional<BibEntry>> fetcherWorker = new FetcherWorker();
+
+    public EntryTypeView(BasePanel basePanel) {
+        this.basePanel = basePanel;
+        this.setTitle(Localization.lang("Select entry type"));
+        //viewModel = new EntryTypeViewModel(basePanel);
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        //ControlHelper.setAction(cancelButton, getDialogPane(), event -> cancelHandle(event));
+        cancelButton.setOnAction(event -> cancelHandle(event));
+
+        //cancelButton.getInputMap(JComponent.WHEN_IN_FOCUSED_WINDOW).put(Globals.getKeyPrefs().getKey(KeyBinding.CLOSE), "close");
+        //cancelButton.getActionMap().put("close", cancelAction);
+        //ControlHelper.setAction(generateButton, getDialogPane(), action -> {
+        //    fetcherWorker.run();
+        //});
+        generateButton.setOnAction(action -> {
+            fetcherWorker.run();
+        });
+        comboBox.onActionProperty().addListener(e -> {
+            idTextField.requestFocus();
+            idTextField.selectAll();
+        });
+
+        idTextField.onActionProperty().addListener(event -> fetcherWorker.run());
+
+        if (basePanel.getBibDatabaseContext().isBiblatexMode()) {
+            biblatexPane.setContent(createPane(BiblatexEntryTypes.ALL));
+            vBox.getChildren().remove(bibTexPane);
+            vBox.getChildren().remove(ieeetranPane);
+            List<EntryType> customTypes = EntryTypes.getAllCustomTypes(BibDatabaseMode.BIBLATEX);
+            if (EntryTypes.getAllCustomTypes(BibDatabaseMode.BIBLATEX).isEmpty()) {
+                vBox.getChildren().remove(customPane);
+            } else {
+                customPane.setContent(createPane(customTypes));
+            }
+            //vBox.getChildren().remove(customPane);
+            //bibTexPane.setVisible(false);
+            //ieeetranPane.setVisible(false);
+        } else {
+            bibTexPane.setContent(createPane(BibtexEntryTypes.ALL));
+            ieeetranPane.setContent(createPane(IEEETranEntryTypes.ALL));
+            vBox.getChildren().remove(biblatexPane);
+
+            List<EntryType> customTypes = EntryTypes.getAllCustomTypes(BibDatabaseMode.BIBTEX);
+            if (EntryTypes.getAllCustomTypes(BibDatabaseMode.BIBTEX).isEmpty()) {
+                vBox.getChildren().remove(customPane);
+            } else {
+                customPane.setContent(createPane(customTypes));
+            }
+            //vBox.getChildren().remove(customPane);
+            //biblatexPane.setVisible(false);
+        }
+
+    }
+
+    private GridPane createPane(Collection<? extends EntryType> entries) {
+        GridPane gridpane = new GridPane();
```

Comment:
It is probably better to use a [FlowPane](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/layout/FlowPane.html) which automatically takes care of organizing its children in columns and rows depending on the preferred width.